server/routerlicious/packages/services-shared/src/runner.ts

```diff
@@ -60,22 +60,43 @@ export function runService<T extends IResources>(
         ? nconf.argv().env({ separator: "__", parseValues: true }).file(configOrPath).use("memory")
         : configOrPath;
 
+    const runnerMetric = Lumberjack.newLumberMetric(LumberEventName.RunService);
     const runningP = run(config, resourceFactory, runnerFactory, logger);
 
     runningP.then(
-        () => {
-            logger?.info("Exiting");
-            Lumberjack.info("Exiting");
+        async () => {
+            await executeAndWait(
+                () => {
+                    logger?.info("Exiting");
+                    runnerMetric.success(`${group} exiting.`);
+                },
+                1000);
             process.exit(0);
         },
-        (error) => {
-            logger?.error(`${group} service exiting due to error`);
-            Lumberjack.error(`${group} service exiting due to error`, undefined, error);
-            logger?.error(inspect(error));
+        async (error) => {
+            await executeAndWait(
```

Comment:
@tanviraumi @znewton Remember we once discussed about missing logs indicating the service crashed/exited? We observed that when using `logger`/`winston` since the Node.js process would exit before `winston` would send the data through all the transports. I observed a similar behavior with `Lumberjack` during my tests for this PR, where I'm introducing a new `runService` metric. During my tests, I noticed the Node.js process only needs about 6-10 ms of extra time to be able to flush telemetry through `winston`/`Lumberjack`. And that is only required when we make the process exit, which happens here in `runService`. So I thought: the easiest approach to solve the problem is introducing a short "wait" before we exit, after we have produced all the telemetry we need. Especially since the wait is only going to happen when we want to exit/crash (new requests/ops do not matter anymore), so we can probably afford to wait 1 second before crashing. Let me know what you think!